src/models/metrics-table.config.ts

```diff
@@ -0,0 +1,369 @@
+        const elements: AggregationResponse[] = [];
+        keys.forEach(k => {
+            const ar: AggregationResponse = {
+                key: k,
+                key_as_string: k,
+                count: 0
+            };
+            elements.push(ar);
+        });
+        return elements;
+    }
+}
+
+export interface MetricsTableColumn {
+    collection: string;
+    termfield: string;
+    metric: ArlasApiMetric.CollectFctEnum | 'count';
+    field?: string;
+}
+export interface MetricsTableConfig {
+    [collection_termfield: string]: MetricsVectorConfig;
+}
+
+export interface MetricsVectorConfig {
+    termfield: string;
+    collection: string;
+    metrics: MetricConfig[];
+}
+
+export interface MetricConfig {
+    field: string;
+    metric: ArlasApiMetric.CollectFctEnum | 'count';
+}
+
+export interface MetricsTableSortConfig {
+    collection: string;
+    termfield: string;
+    order: 'asc' | 'desc';
+    on: 'alphabetical' | 'count' | 'metric';
+    metric?: MetricConfig;
+}
+
+
+/**
+ * |                             MetricsVectors                            |
+ * |        |      MetricVector 1        |          MetricVector 2         |
+ * |        | (c1,f1,m1)  |  (c1,f2,m2)  |  (c2,f'1,m'2)  |  (c2,f'2,m'2)  |
+ * | term 1 |      x      |      x       |      x         |      x         |
+ * | term 2 |      x      |      x       |      x         |      x         |
+ */
+
+export class MetricsVectors {
+    public collections_fields: Set<string> = new Set();
+    /** A vector represents a collection */
+    public vectors: MetricsVector[] = [];
+    public constructor(config: MetricsTableConfig, sortConfig: MetricsTableSortConfig, nbTerms: number) {
+        Object.keys(config).forEach(collection_fields => {
+            const collectionfieldConfig = config[collection_fields];
+            const uniqueConfigKey = collectionfieldConfig.collection + collectionfieldConfig.termfield;
+            if (collection_fields !== uniqueConfigKey) {
+                // throw error ? we need to be sure that the key is always built the same way.
+            }
+            if (!this.collections_fields.has(uniqueConfigKey)) {
+                this.collections_fields.add(uniqueConfigKey);
+                this.vectors.push(new MetricsVector(
+                    collectionfieldConfig.collection,
+                    collectionfieldConfig.termfield , collectionfieldConfig, sortConfig, nbTerms));
+            }
+        });
+    }
+
+}
+
+/** @class */
+/** A MetricsVector reprensents a collection with all its metrics.
+ * It provides `getAggregation`method that returns the arlas-api Aggregation request object.
+ * |        |      MetricVector 1        |
+ * |        | (c1,f1,m1)  |  (c1,f2,m2)  |
+ * | term 1 |      x      |      x       |
+ * | term 2 |      x      |      x       |
+ */
+export class MetricsVector {
+    public collection: string;
+    public termfield: string;
```

Comment:
Camel case ?

Reply:
This might sound crazyn but there are some attribute where for the sobriety i preferred noting it as a one word: termfiled, rowfield